common/common.go

```diff
@@ -65,7 +65,8 @@ func (h DemoHeader) TickTime() time.Duration {
 // positions between the time at which they are thrown and until they detonate.
 type GrenadeProjectile struct {
 	EntityID   int
-	Weapon     EquipmentElement
+	// IMPORTANT: I Switched Weapon from EquipmentElement to Equipment type, this means it could break user's code (feel free to do otherwise)...
+	Weapon     Equipment
```

Comment:
Revert due to incompatibility. Maybe we can add a new field `WeaponInstance`  of type `Equipment` if it's needed

Reply:
Right, `WeaponInstance` seems a good name to me.

However I don't know if you wish to handle deprecation but we might declare `Weapon` as deprecated for a futur major version ? (because it's kinda a duplicate field)

Reply:
First we need to make sure whether `WeaponInstance` can be nil or not.

If it can't be nil we can put a comment behind `Weapon` saying `Deprecated, use WeaponInstance.Weapon` instead.

Reply:
ah just saw you added the comment already, looks good!